### PR TITLE
fix Azure Vault Json Provider handling of key

### PR DESCRIPTION
**Pull request by norah-li**
Fix Azure Vault JSON Provider handling of key in case of payload with multiple keys. 
Will add tests to the pipeline after secrets are added to the repository.

URL format:
jdbc:oracle:thin:@config-azurevault://{your-vault-name}.vault.azure.net/secrets/{secret-name}?key={your-key-name}

This pull request also adds tests with key options (with AzureVaultJsonProvider, OciVaultJsonProvider and OciObjectStorageProvider).

### CODE REVIEW

File: .github/workflows/run-tests.yaml [hunk 1]
```diff
@@ -75,7 +77,11 @@ jobs:
           OCI_DB_TOOLS_CONNECTION_OCID_PKCS12=${{ secrets.TEST_OCI_DB_TOOLS_CONNECTION_OCID_PKCS12 }}\n
           OCI_DB_TOOLS_CONNECTION_OCID_SSO=${{ secrets.TEST_OCI_DB_TOOLS_CONNECTION_OCID_SSO }}\n
           OCI_OBJECT_STORAGE_URL=${{ secrets.TEST_OCI_OBJECT_STORAGE_URL }}\n
+          OCI_OBJECT_STORAGE_URL_MULTIPLE_KEYS=${{ secrets.TEST_OCI_OBJECT_STORAGE_URL_MULTIPLE_KEYS }}\n
+          OCI_OBJECT_STORAGE_URL_KEY=${{ secrets.TEST_OCI_OBJECT_STORAGE_URL_KEY }}\n
           OCI_PASSWORD_PAYLOAD_OCID=${{ secrets.TEST_OCI_PASSWORD_PAYLOAD_OCID }}\n
+          OCI_PASSWORD_PAYLOAD_OCID_MULTIPLE_KEYS=${{ secrets.TEST_OCI_PASSWORD_PAYLOAD_OCID_MULTIPLE_KEYS }}\n
```
Comment (psilberk) [inline .github/workflows/run-tests.yaml:83]: Are we missing a test with OCIVaultJsonProvider (with and without multiple keys)?

File: ojdbc-provider-oci/src/test/java/oracle/jdbc/provider/oci/configuration/OciVaultJsonProviderTest.java [hunk 2]
```diff
@@ -28,23 +28,38 @@ public class OciVaultJsonProviderTest {
    * Verifies the AUTHENTICATION=OCI_DEFAULT parameter setting.
    * This test will fail if (~/.oci/config) or (~/.oraclebmc/config) or the
    * environmental variable OCI_CONFIG_FILE is not set.
-   */
+   **/
```
Comment (psilberk) [inline ojdbc-provider-oci/src/test/java/oracle/jdbc/provider/oci/configuration/OciVaultJsonProviderTest.java:31]: Please remove this closing extra * (it's on several places)

**Review by psilberk — COMMENTED** [review]
Please see comments

File: .github/workflows/run-tests.yaml (hunk 1 above)
Comment (norah-li) [reply .github/workflows/run-tests.yaml:83]: Yes, the existing tests are without key options.
This pull request adds tests with key options (with AzureVaultJsonProvider, OciVaultJsonProvider and OciObjectStorageProvider). I'll update this in description.

**Review by jeandelavarene — APPROVED** [review]

**Review by ting-lan-wang — APPROVED** [review]

**Review by psilberk — APPROVED** [review]

**Pull request merged**